doc/md/go-api/inspection.md

```diff
@@ -0,0 +1,79 @@
+---
+title: Inspecting Schemas 
+slug: /go-api/inspection
+---
+## Inspection
+
+Inspection is the one of Atlas's core capabilities. When we say "inspection" in the context of this project we mean the
+process of connecting to an existing database, querying its metadata tables to understand the structure of the
+different tables, types, extensions, etc.
+
+Databases vary greatly in the API they provide users to understand a specific database's schema, but Atlas goes to great
+lengths to abstract these differences and provide a unified API for inspecting databases. Consider the `Inspector`
+interface in the [sql/schema](https://pkg.go.dev/ariga.io/atlas@v0.3.2/sql/schema#Inspector)
+package:
+
+```go
+// Inspector is the interface implemented by the different database
+// drivers for inspecting multiple tables.
+type Inspector interface {
+  // InspectSchema returns the schema description by its name. An empty name means the
+  // "attached schema" (e.g. SCHEMA() in MySQL or CURRENT_SCHEMA() in PostgreSQL).
+  // A NotExistError error is returned if the schema does not exists in the database.
+  InspectSchema(ctx context.Context, name string, opts *InspectOptions) (*Schema, error)
+  
+  // InspectTable returns the table description by its name. A NotExistError
+  // error is returned if the table does not exists in the database.
+  InspectTable(ctx context.Context, name string, opts *InspectTableOptions) (*Table, error)
+  
+  // InspectRealm returns the description of the connected database.
+  InspectRealm(ctx context.Context, opts *InspectRealmOption) (*Realm, error)
+}
+```
```

Comment:
Add link to godoc. I'm going to drop the inspect-table one and we don't want to maintain it twice.

Reply:
See L12 for godoc link :-)